lib/std/tar.zig

```diff
@@ -255,6 +263,30 @@ pub fn pipeToFileSystem(dir: std.fs.Dir, reader: anytype, options: Options) !voi
                     }
                 }
             },
+            .GNUTYPE_LONGNAME => {
+                assert(std.mem.eql(u8, header.name(), "././@LongLink"));
+
+                while (true) {
+                    const filename_chunk = try buffer.readChunk(reader, 512);
```

Comment:
You only support longname fields up to 512 bytes, but the length of the longname is defined in the header, and can be >512.

Reply:
i'll try it, thanks. Its just what I came up with thinking not advancing the header buffer may cause issues

Reply:
after looking into it, one reason to do it this way could be if we barely reach MAX_PATH_BYTES, the 512 byte padding or next chunk could overflow the filename buffer.
It is highly unlikely though, as MAX_PATH_BYTES  probably is always a multiple of 512, only a path with exactly MAX_PATH_BYTES will need to read another chunk from the stream to find the null terminator and overflow the buffer

Reply:
The unpaddded length of the longname is given in the header, if it's more than MAX_PATH_BYTES, throw an error.